classes/profile_helper.php

```diff
@@ -104,51 +119,34 @@ public static function get_min_duration_for_group_and_reason(string $group, int
      * @return float duration (as seconds) of the fastest profile for a given reason.
      */
     public static function get_min_duration_for_reason(int $reason, bool $usecache = true): float {
-        global $DB;
+        $quota = self::$quotas[$reason];
 
-        $reasonstr = profile::REASON_STR_MAP[$reason];
-        $quota = (int) get_config('tool_excimer', "num_$reasonstr");
-
-        $cachekey = 'profile_type_' . $reason . '_min_duration_s';
-        $result = get_config('tool_excimer', $cachekey);
+        $cachekey = '__all__';
+        $cachefield = 'profile_type_' . $reason . '_min_duration_s';
+        $cache = \cache::make('tool_excimer', 'request_metadata');
+        $result = $cache->get($cachekey);
 
-        if (!$usecache || $result === false) {
+        if (!$usecache || $result === false || !isset($result[$cachefield])) {
             // Get and set cache.
-            $reasons = $DB->sql_bitand('reason', $reason);
+            $db = manager::get_altconnection();
+            $reasons = $db->sql_bitand('reason', $reason);
             $sql = "SELECT duration as min_duration
                       FROM {tool_excimer_profiles}
                      WHERE $reasons != ?
                   ORDER BY duration DESC
                      ";
-            $resultset = $DB->get_records_sql($sql, [
+            $resultset = $db->get_records_sql($sql, [
                 profile::REASON_NONE,
             ], $quota - 1, 1); // Will fetch the Nth item based on the quota.
             // Cache the results in (avoids recalculation later).
             $newvalue = (end($resultset)->min_duration ?? 0.0);
             // Updates the cache value if the calculated value is different.
-            if ($result !== $newvalue) {
-                $result = $newvalue;
-                set_config($cachekey, $result, 'tool_excimer');
-            }
-        }
-        return (float) $result;
-    }
-
-    /**
-     * Clears the plugin cache for keys used for the provided reasons
-     *
-     * @author Kevin Pham <kevinpham@catalyst-au.net>
-     *
-     * @param int $reason bitmap of reason(s)
-     */
-    public static function clear_min_duration_cache_for_reason(int $reason): void {
```

Comment:
is it worth keeping this method but just changing its implementation?

Reply:
The delete statements that replace it are all one-liners.

I noticed that the way the cache was implemented is sub-optimal, and want to create a separate issue for it. For the purpose of this bugfix, however, this is sufficient.

Reply:
ok cool